### PR TITLE
VIH-7354 alert for network poor consultation

### PR DESCRIPTION
**Pull request by valentinabennett**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/VIH-7354

### Change description ###
Fix alert for poor network connection

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/services/notification-toastr.service.spec.ts [hunk 1]
```diff
@@ -261,15 +261,11 @@ describe('NotificationToastrService', () => {
...
-        service.reportPoorConnection(
-            new ParticipantHeartbeat(globalConference.id, globalParticipant.id, HeartbeatHealth.Poor, '', '', '', '')
-        );
-        service.reportPoorConnection(
-            new ParticipantHeartbeat(globalConference.id, globalParticipant.id, HeartbeatHealth.Poor, '', '', '', '')
-        );
-        service.reportPoorConnection(
-            new ParticipantHeartbeat(globalConference.id, globalParticipant.id, HeartbeatHealth.Poor, '', '', '', '')
-        );
+        for (let i = 0; i < 26; i++) {
```
Comment (levibutler) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/services/notification-toastr.service.spec.ts:264]: can we test that 27 returns 2?

Comment (levibutler) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/services/notification-toastr.service.spec.ts:264]: also tied to the interval that may change or be unreliable, could we use a timer instead to reset? and use ticks(120000) in the test?

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/services/notification-toastr.service.ts [hunk 2]
```diff
@@ -108,7 +108,12 @@ export class NotificationToastrService {
     reportPoorConnection(heartbeat: ParticipantHeartbeat) {
         const heartbeatKey = `${heartbeat.participantId}_${heartbeat.heartbeatHealth.toString()}`;
         if (this.activeHeartbeatReport.indexOf(heartbeatKey) >= 0) {
-            return;
+            this.activeHeartbeatReport.push(heartbeatKey);
+            if (this.activeHeartbeatReport.filter(x => x.indexOf(heartbeatKey) >= 0).length > 25) {
```
Comment (levibutler) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/services/notification-toastr.service.ts:112]: can we setup a timer on 2 min interval to clear the array?

Comment (valentinabennett) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/services/notification-toastr.service.ts:112]: done

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/services/notification-toastr.service.spec.ts (hunk 1 above)
Comment (valentinabennett) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/services/notification-toastr.service.spec.ts:264]: not the case, should be one

**Review by levibutler — APPROVED** [review]
Looks good, suggested move to a timer instead of counting the number of heartbeats but looks like it was rejected. Still happy it will do the job. Approved

**Pull request merged**